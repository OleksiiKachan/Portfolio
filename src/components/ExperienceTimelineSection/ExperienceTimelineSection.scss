.experienceTimelineSection {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.experienceTimelineSection * {
    box-sizing: border-box;
}

.experienceTimelineSection__title {
    margin-bottom: 15px;
}

.experienceTimelineSection__timeline {
	width: 90%;
    padding: 25px 0;
    position: relative;
    max-width: 1200px;
    margin: 0 auto;
    color: white;
}

.experienceTimelineSection__timeline::after {
    content: '';
    position: absolute;
    width: 6px;
    background-color: #474e5d;
    top: 0;
    bottom: 0;
    left: 50%;
    margin-left: -3px;
}

.timelineItem__container {
    padding: 10px 40px;
    position: relative;
    background-color: inherit;
    width: 50%;
}

.timelineItem__container::after {
    content: '';
    position: absolute;
    width: 25px;
    height: 25px;
    right: -17px;
    background-color: rgb(187, 45, 45);
    border: 4px solid #FF9F55;
    top: 15px;
    border-radius: 50%;
    z-index: 1;
}

.timelineItem__container_left {
    left: 0;
}

.timelineItem__container_right {
    left: 50%;
}

.timelineItem__container_left::before {
    content: " ";
    height: 0;
    position: absolute;
    top: 22px;
    width: 0;
    z-index: 1;
    right: 30px;
    border: medium solid #474e5d;
    border-width: 10px 0 10px 10px;
    border-color: transparent transparent transparent #474e5d;
}

.timelineItem__container_right::before {
    content: " ";
    height: 0;
    position: absolute;
    top: 22px;
    width: 0;
    z-index: 1;
    left: 30px;
    border: medium solid #474e5d;
    border-width: 10px 10px 10px 0;
    border-color: transparent #474e5d transparent transparent;
}

.timelineItem__container_right::after {
    left: -16px;
}

.timelineItem__content {
    padding: 20px 30px;
    background-color: #474e5d;
    position: relative;
    border-radius: 6px;
}

.timelineItem__description {
    font-size: smaller;
}

@media screen and (max-width: 768px) {
    .experienceTimelineSection__timeline::after {
        left: 31px;
    }

    .timelineItem__container {
        width: 100%;
        padding-left: 70px;
        padding-right: 25px;
    }

    .timelineItem__container::before {
        left: 60px;
        border: medium solid white;
        border-width: 10px 10px 10px 0;
        border-color: transparent white transparent transparent;
    }

    .timelineItem__container_left::after, .timelineItem__container_right::after {
        left: 15px;
    }

    .timelineItem__container_right {
        left: 0%;
    }
}