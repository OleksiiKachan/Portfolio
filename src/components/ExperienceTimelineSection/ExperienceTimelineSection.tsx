import React, { Component } from 'react';
import classNames from 'classnames';
import './ExperienceTimelineSection.scss';

/*
    <ExperienceTimelineSection timeline={}/>
*/
class ExperienceTimelineSection extends Component<any, any> {
  previousSide: string;

  constructor(props: any) {
    super(props);

    this.previousSide = 'right';
  }

  render() {
    return (
      <div
        id="education"
        className={classNames(
          'experienceTimelineSection',
          this.props.className
        )}
      >
        <h2 className={classNames('experienceTimelineSection__title')}>
          Education & Experience
        </h2>
        <div className={classNames('experienceTimelineSection__timeline')}>
          {this.props.timeline.map(
            (timelineItem: {
              time: string;
              title: string;
              description: string;
            }) => {
              const sideClass =
                this.previousSide === 'right'
                  ? 'timelineItem__container_left'
                  : 'timelineItem__container_right';
              this.previousSide =
                this.previousSide === 'right' ? 'left' : 'right';
              return (
                <div
                  className={classNames('timelineItem__container', sideClass)}
                >
                  <div className={classNames('timelineItem__content')}>
                    <h3 className={classNames('timelineItem__time')}>
                      {timelineItem.time}
                    </h3>
                    <h4 className={classNames('timelineItem__title')}>
                      {timelineItem.title}
                    </h4>
                    <p className={classNames('timelineItem__description')}>
                      {timelineItem.description}
                    </p>
                  </div>
                </div>
              );
            }
          )}
        </div>
      </div>
    );
  }
}

export default ExperienceTimelineSection;
