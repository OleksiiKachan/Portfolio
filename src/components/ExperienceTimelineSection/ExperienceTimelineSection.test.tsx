import React from 'react';
import { shallow } from 'enzyme';
import ExperienceTimelineSection from './';

describe('ExperienceTimelineSection', () => {
  it('Snapshot', () => {
    const timeline = [
      {
        title: 'Title-education-left',
        subtitle: 'Subtitle-education-left',
        dates: 'Dates-education-left',
        type: 'education',
      },
      {
        title: 'Title-education-right',
        subtitle: 'Subtitle-education-right',
        dates: 'Dates-education-right',
        type: 'education',
      },
      {
        title: 'Title-work-left',
        subtitle: 'Subtitle-work-left',
        dates: 'Dates-work-left',
        type: 'work',
      },
      {
        title: 'Title-work-right',
        subtitle: 'Subtitle-work-right',
        dates: 'Dates-work-right',
        type: 'work',
      },
    ];
    const component = shallow(
      <ExperienceTimelineSection timeline={timeline} />
    );
    expect(component).toMatchSnapshot();
  });
});
