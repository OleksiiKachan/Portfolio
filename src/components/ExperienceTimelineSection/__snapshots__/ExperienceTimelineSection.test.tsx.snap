// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceTimelineSection Snapshot 1`] = `
<div
  className="experienceTimelineSection"
  id="education"
>
  <h2
    className="experienceTimelineSection__title"
  >
    Education & Experience
  </h2>
  <div
    className="experienceTimelineSection__timeline"
  >
    <_default
      index={0}
      item={
        Object {
          "dates": "Dates-education-left",
          "subtitle": "Subtitle-education-left",
          "title": "Title-education-left",
          "type": "education",
        }
      }
      key="0"
      side="left"
    />
    <_default
      index={1}
      item={
        Object {
          "dates": "Dates-education-right",
          "subtitle": "Subtitle-education-right",
          "title": "Title-education-right",
          "type": "education",
        }
      }
      key="1"
      side="right"
    />
    <_default
      index={2}
      item={
        Object {
          "dates": "Dates-work-left",
          "subtitle": "Subtitle-work-left",
          "title": "Title-work-left",
          "type": "work",
        }
      }
      key="2"
      side="left"
    />
    <_default
      index={3}
      item={
        Object {
          "dates": "Dates-work-right",
          "subtitle": "Subtitle-work-right",
          "title": "Title-work-right",
          "type": "work",
        }
      }
      key="3"
      side="right"
    />
  </div>
</div>
`;
